tests/autorestart_test/test_container_autorestart.py

```diff
@@ -0,0 +1,327 @@
+            critical_group_list.append(identifier_value)
+        elif identifier_key == "program" and identifier_value:
+            critical_process_list.append(identifier_value)
+        else:
+            pytest_assert(False, "Syntax of the line {} in critical_processes \
+                file is incorrect.".format(line))
+ 
+    return critical_group_list, critical_process_list
+
+def get_group_program_info(duthost, container_name, critical_group):
+    """
+    @summary: Get critical program names, running status and their pids by analyzing the command
+              output of "docker exec <container_name> supervisorctl"
+    @return: A dictionary where keys are the critical program names and values
+             are their running status and pids
+    """
+    group_program_info = defaultdict(list)
+    program_name = None
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(critical_group) != -1:
+            program_name = program_info.split()[0].split(':')[1].strip(' \n')
+            program_status = program_info.split()[1].strip()
+            if program_status in ["EXITED", "STOPPED", "STARTING"]:
+                program_pid = -1
+            else:
+                program_pid = program_info.split()[3].strip(' ,')
+
+            group_program_info[program_name].append(program_status)
+            group_program_info[program_name].append(program_pid)
+            
+    return group_program_info
+
+def get_program_info(duthost, container_name, program_name):
+    """
+    @summary: Get program running status and its pid by analyzing the command
+              output of "docker exec <container_name> supervisorctl"
+    @return:  Program running status and its pid
+    """
+    program_status = None
+    program_pid = -1
+
+    program_list = duthost.shell("docker exec {} supervisorctl".format(container_name))
+    for program_info in program_list["stdout_lines"]:
+        if program_info.find(program_name) != -1:
+            program_status = program_info.split()[1].strip()
+            if program_status == "RUNNING":
+                program_pid = int(program_info.split()[3].strip(' ,'))
+            break
+
+    if program_status:
+        logger.info("Found program {} in the {} state with pid {}".format(program_name, program_status, program_pid))
+            
+    return program_status, program_pid
+
+def get_autorestart_container_and_state(duthost):
+    """
+    @summary: Get container names and its autorestart states by analyzing 
+              the command output of "show container feature autorestart"
+    @return:  A dictionary where keys are the names of containers which have the 
+              autorestart feature implemented and values are the autorestart feature
+              state for that container
+    """
+    autorestart_containers = {}
```

Comment:
This variable name is vague and confusing. It should contain mention of "state", like `container_autorestart_states`